src/main/scala/is/hail/methods/LDMatrix.scala

```diff
@@ -54,18 +49,23 @@ object LDMatrix {
       val localMat: DenseMatrix = normalizedBlockMatrix.toLocalMatrix().asInstanceOf[DenseMatrix]
       val product = localMat multiply localMat.transpose
       indexedRowMatrix =
-        BlockMatrixIsDistributedMatrix.from(normalizedBlockMatrix.blocks.sparkContext, product,
-          normalizedBlockMatrix.rowsPerBlock, normalizedBlockMatrix.colsPerBlock).toIndexedRowMatrix()
+        BlockMatrixIsDistributedMatrix.from(vds.sparkContext, product, normalizedBlockMatrix.rowsPerBlock,
+          normalizedBlockMatrix.colsPerBlock).toIndexedRowMatrix()
     }
     else {
-      indexedRowMatrix = (normalizedBlockMatrix multiply normalizedBlockMatrix.transpose)
+      import is.hail.distributedmatrix.DistributedMatrix.implicits._
+      val dm = DistributedMatrix[BlockMatrix]
+      import dm.ops._
+      normalizedBlockMatrix.persist(StorageLevel.MEMORY_AND_DISK)
+      normalizedBlockMatrix.blocks.count()
```

Comment:
It looks like you're counting to cache the data before the next line, but I don't think count will accomplish that.

`def count(): Long = sc.runJob(this, Utils.getIteratorSize _).sum`